src/components/application_manager/test/commands/mobile/get_system_capability_request_test.cc

```diff
@@ -0,0 +1,246 @@
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "application_manager/commands/mobile/get_system_capability_request.h"
+#include "application_manager/test/include/application_manager/commands/command_request_test.h"
+#include "application_manager/test/include/application_manager/mock_hmi_capabilities.h"
+#include "gtest/gtest.h"
+
+namespace test {
+namespace components {
+namespace commands_test {
+namespace mobile_commands_test {
+namespace get_vehicle_data_tequest {
+
+using ::testing::_;
+using ::testing::Return;
+using ::testing::ReturnRef;
+namespace am = ::application_manager;
+namespace amt = application_manager_test;
+using am::commands::GetSystemCapabilityRequest;
+using am::commands::MessageSharedPtr;
+using am::event_engine::Event;
+namespace mobile_result = mobile_apis::Result;
+
+namespace {
+const uint32_t kConnectionKey = 2u;
+}  // namespace
+
+class GetSystemCapabilityRequestTest
+    : public CommandRequestTest<CommandsTestMocks::kIsNice> {
+ public:
+  MessageSharedPtr CreateMessageParams() {
+    MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+    (*msg)[am::strings::params][am::strings::connection_key] = kConnectionKey;
+    return msg;
+  }
+};
+
+TEST_F(GetSystemCapabilityRequestTest, Run_NotExistApp_AppNotRegistered) {
+  MockAppPtr mock_app;
+
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+  MessageSharedPtr msg = CreateMessageParams();
+
+  SharedPtr<GetSystemCapabilityRequest> command(
+      CreateCommand<GetSystemCapabilityRequest>(msg));
+
+  EXPECT_CALL(
+      app_mngr_,
+      ManageMobileCommand(
+          MobileResultCodeIs(mobile_apis::Result::APPLICATION_NOT_REGISTERED),
+          _)).WillOnce(Return(true));
+
+  ASSERT_TRUE(command->Init());
+  command->Run();
+}
+
+TEST_F(GetSystemCapabilityRequestTest, Run_EmptyMessage_ExpectInvalidData) {
+  MockAppPtr mock_app = CreateMockApp();
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+
+  MessageSharedPtr msg = CreateMessageParams();
+  SharedPtr<GetSystemCapabilityRequest> command(
+      CreateCommand<GetSystemCapabilityRequest>(msg));
+
+  EXPECT_CALL(app_mngr_,
+              ManageMobileCommand(
+                  MobileResultCodeIs(mobile_apis::Result::INVALID_DATA), _))
+      .WillOnce(Return(true));
+
+  ASSERT_TRUE(command->Init());
+  command->Run();
+}
+
+TEST_F(GetSystemCapabilityRequestTest, Run_NAVIGATION_SUCCESS) {
+  MockAppPtr mock_app = CreateMockApp();
+  SmartObject sm(smart_objects::SmartType_Map);
```

Comment:
@ValeriiMalkov `sm` unclear name.

Reply:
@BSolonenko fixed at the commit ab40e21d3a66941f9a9d9d0a0eac8e18f09add44